tests/xharness/GitHub.cs

```diff
@@ -40,5 +40,73 @@ public static IEnumerable<string> GetLabels (Harness harness, int pull_request)
 
 			return File.ReadAllLines (path);
 		}
+
+		public static IEnumerable<string> GetModifiedFiles (Harness harness, int pull_request)
+		{
+			var path = Path.Combine (harness.LogDirectory, "pr" + pull_request + "-files.log");
+			if (!File.Exists (path)) {
+				Directory.CreateDirectory (harness.LogDirectory);
+				using (var client = new WebClient ()) {
```

Comment:
That doesn't work with PRs with multiple commits, in which case you'll need the hash for the first commit.

It doesn't look like jenkins provides this information, and it's not available in the git repo as far as I can tell.

It is however available by asking GitHub for information about the pull request, so there would be at least one network request.